roles/setup_repo/tasks/PG_RedHat_setuprepos.yml

```diff
@@ -92,8 +92,43 @@
 # Executing yum/dnf updateinfo -y makes the import of GPG keys effective even
 # if command's return is different from 0.
 - name: Execute yum updateinfo
-  ansible.builtin.command: yum updateinfo -y
+  ansible.builtin.shell: >
+    set -o pipefail;
+    yum updateinfo -y
   register: updateinfo_output
   become: true
   failed_when: false
   changed_when: updateinfo_output.rc != 0
+
+# Adding BDR repository if BDR nodes are defined
+- name: Install curl package if not available
+  ansible.builtin.package:
+    name: curl
+    state: present
+  become: yes
+
+- name: Install BDR package if BDR is enabled
+  ansible.builtin.shell: >
+      set -o pipefail;
+      curl -sS "{{ edb_2q_base_repo_link }}/bdr4/release/{{ pg_version }}/rpm" | bash
+  args:
+    executable: /bin/bash
+  register: reposub
+  failed_when: >
+    reposub.rc != 0 or 'error: ' in reposub.stdout.lower()
+  when: >
+    bdr_nodes_ips is defined and ansible_host in bdr_nodes_ips
+    and pg_version|int >= 13
+
+- name: Install HARP package if BDR is enabled
+  ansible.builtin.shell: >
+      set -o pipefail;
+      curl -sS "{{ edb_2q_base_repo_link }}/harp/release/{{ pg_version }}/rpm" | bash
+  args:
+    executable: /bin/bash
+  register: reposub
+  failed_when: >
+    reposub.rc != 0 or 'error: ' in reposub.stdout.lower()
+  when: >
+    bdr_nodes_ips is defined and ansible_host in bdr_nodes_ips
+    and pg_version|int >= 13
```

Comment:
What do think you about using a new variable containing the list of repositories and looping over this list:
```yaml
2nq_repositories:
- bdr/release
- harp/release
```

Reply:
Modified as per your suggestion. Please have a look.